Quickstart (#22)
castelao/NREL-pypackage-template: README.md

# Template for NREL's Python packages

This is an opinionated template for creating Python packages at NREL,
intended for a wide range of users. If you are new to NREL or don't have
much experience building Python packages, this template is a good option

> to provide you a scaffolding to get started quickly. It also setup several
> support tools such checks on your code, documentation generation, and
> running tests.

Reviewer: ```suggestion
to provide you a scaffolding to get started quickly. It also sets up several
support tools for you, such as automated checks on your code, automatic 
documentation generation, and tests validation on every pull request.
```

> documentation generation, and tests validation on every pull request.

Reviewer 2: I think we need to add a description of all support tools that are included in this template somewhere (not for the Quickstart, though).

Author: I opened an issue #24 so we don't forget about that.

If you're an experienced Python developer, you still have some control by

> answering a few questions during the initialization process, while avoiding
> a plain copy and paste from some of your other projects. We use this

Reviewer: ```suggestion
answering a few questions during the initialization process, which helps avoid
a plain copy and paste from your other projects. We use this
```

  ```

We use pixi for development. If you would like to contribute to this repository or want to build your own template, we strongly recommend using Pixi.

#### uv

If you prefer using [uv](https://docs.astral.sh/uv/#tool-management), you can install `copier` with the following command:

```bash
uv tool install copier
```

Note: If you use `poetry`, you might want to consider moving to `uv`.

#### pipx

If you have [pipx](https://pypa.github.io/pipx/) installed, you can install `copier` with the following command:

```bash
pipx install copier
```

#### Nix flake

If you are using [Nix flakes](https://nixos.wiki/wiki/Flakes), you can install `copier` with the following command:

```bash
nix profile install 'https://flakehub.com/f/copier-org/copier/*.tar.gz'
```

### 2) Build your package

```bash
copier copy https://github.com/castelao/NREL-pypackage-template ../my_new_project
cd ../my_new_project
```

You should see several files and directories created from the information
that you gave. For instance, take a look at the `pyproject.toml` file.
Your package will probably depend on other libraries. Let's assume that
you'll use `xarray`, so you can run
```bash
pixi add xarray

> ```

Reviewer: This depends on using pixi, right? Is it worth adding instructions if users are using homebrew or pipx or something else instead?

Author: Good catch. The project template is ready for pixi and assume one would use it, but conflicts with the previous instructions by allowing install with other tools. Shall we just assume that everyone should use pixi and simplify the instructions? 1) install pixi 2) create an environment to install copier 3) copy template.

Reviewer: I'm of course happy with that, but would that potentially prevent more people from using it? Or are we not targeting people with an opinion around that anyways?

Reviewer 2: I particularly think that it would be better to just assume the user will use `pixi`. This is also a lot easier to keep instructions up to date.

Author: A compromise here would he to have the QuickStart assuming the environment that we work with, such as `pixi` `GitHub`, .... And in the documentation expand for other options.

I'll move ahead this PR as it is because the current instructions are quite limited, and later move some stuff to a documentation.

> Which will update the `pyproject.toml` and `pixi.lock` files with the new

Reviewer: ```suggestion
which will update the `pyproject.toml` and `pixi.lock` files with the new
```
